### ISSUE TITLE
Feature Request: Do not warn on assumeNonNull(nullUnmarkedThing)

### ISSUE THREAD

**Issue by agrieve** [uber/NullAway#1241]
E.g.:
```
 warning: [NullAway] passing known @NonNull parameter 'tabGroupModelFilter' to CastToNonNullMethod (org.chromium.build.NullUtil.assumeNonNull) at position 0. This method argument should only take values that NullAway considers @Nullable at the invocation site, but which are known not to be null at runtime.
```

The warning is great for values that are known to be non-null, but for values from methods annotated with `@NullUnmarked`, it would make migrations easier if these warnings would not show (e.g. you could incrementally fix them, and then remove `@NullUnmarked` at the end).

**Comment by raccoonback** [comment]
@agrieve @msridhar 
Hello!
I'm interested in working on this issue.
Would it be possible for me to take it on and start working on it?
Additionally, if there are any guidelines or specific points I should keep in mind while addressing this issue, I'd appreciate it if you could share them.

**Comment by raccoonback** [comment]
@msridhar Sure, I will take this issue and work on it.
